transformations/transformations/pool_allocator.py

```diff
@@ -530,11 +544,29 @@ def inject_pool_allocator_into_calls(self, routine, targets):
         """
         call_map = {}
         stack_var = self._get_local_stack_var(routine)
+        intfs = FindNodes(Interface).visit(routine.spec)
         for call in FindNodes(CallStatement).visit(routine.body):
             if call.name in targets:
+               # If call is declared via an explicit interface, the ProcedureSymbol corresponding to the call is the
```

Comment:
Interesting, I wasn't aware of this. We should keep an eye out for this. I would almost consider this a bug, since the interface is the declaration of the routine and not the actual routine, so getting this after a call to `call.routine` seems counter-intuitive to me. (We should in any case update the interface, of course).

Reply:
I think this behaviour arises from the fact that if we have an interface, then the symbol already exists in the calling subroutine's symbol-table and so `Subroutine.enrich_calls` doesn't overwrite it, even though `update_symbol` is still `True` and we do actually execute `call._update(...)` in `loki/subroutine.py' L455.

I am not familiar at all with the symbol table, but in my mind a clean "fix" could perhaps be to allow entries in the symbol table to be overwritten?